lib/global_id/locator.rb

```diff
@@ -106,7 +106,9 @@ def use(app, locator = nil, &locator_block)
 
       private
         def locator_for(gid)
-          @locators.fetch(normalize_app(gid.app)) { default_locator }
+          @locators.fetch(normalize_app(gid.app)) do
+            gid.respond_to?(:unscoped) ? ACTIVE_RECORD_LOCATOR : DEFAULT_LOCATOR
```

Comment:
also we should use `gid.model_class` or it will always use the `DEFAULT_LOCATOR` as `GlobalID` does not respond to `:unscoped`